### ISSUE TITLE
com.hazelcast.internal.tpcengine.nio.NioAsyncServerSocketTest.test_createCloseLoop_withSameReactor

### ISSUE THREAD

**Issue by sumnerib** [hazelcast/hazelcast#24002]
_master_ (commit 843e71fb7643d5fa775e8145c9edcf559fed2695)

Failed on oracle-17: https://jenkins.hazelcast.com/job/Hazelcast-master-OracleJDK17/73/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/
<details><summary>Stacktrace:</summary>

```
java.io.UncheckedIOException: Failed to bind to /127.0.0.1:5000
	at com.hazelcast.internal.tpcengine.nio.NioAsyncServerSocket.bind(NioAsyncServerSocket.java:106)
	at com.hazelcast.internal.tpcengine.AsyncServerSocket.bind(AsyncServerSocket.java:100)
	at com.hazelcast.internal.tpcengine.AsyncServerSocketTest.test_createCloseLoop_withSameReactor(AsyncServerSocketTest.java:202)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:555)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:337)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294)
	at com.hazelcast.internal.tpcengine.nio.NioAsyncServerSocket.bind(NioAsyncServerSocket.java:104)
	... 42 more

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-OracleJDK17/73/

**Comment by olukas** [comment]
Failed again on _master_ with oracle-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK11/364/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with openjdk-20: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OpenJDK-ea/30/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with oracle-11: https://jenkins.hazelcast.com/job/Hazelcast-master-sonar-new/107/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with redhat-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-RedHatJDK11/81/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by ramizdundar** [comment]
https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/18574/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with openjdk-sts (openjdk-20): https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OpenJDK-sts/42/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with oracle-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK11/374/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _master_ with redhat-11: https://jenkins.hazelcast.com/job/Hazelcast-master-RedHatJDK11/86/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _5.3.z_ with redhat-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-RedHatJDK17/68/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with oracle-11 (little-endian): https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-little-endian/258/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with microsoft-11: https://jenkins.hazelcast.com/job/Hazelcast-master-MicrosoftJDK11/70/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by ihsandemir** [comment]
Failed again on [PR build ](https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/19073/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/).

```
java.io.UncheckedIOException: Failed to bind to /127.0.0.1:5001
	at com.hazelcast.internal.tpcengine.nio.NioAsyncServerSocket.bind(NioAsyncServerSocket.java:106)
	at com.hazelcast.internal.tpcengine.net.AsyncServerSocket.bind(AsyncServerSocket.java:102)
	at com.hazelcast.internal.tpcengine.net.AsyncServerSocketTest.test_createCloseLoop_withSameReactor(AsyncServerSocketTest.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:49)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:120)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:95)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:69)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:459)
	at java.base/sun.nio.ch.Net.bind(Net.java:448)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at com.hazelcast.internal.tpcengine.nio.NioAsyncServerSocket.bind(NioAsyncServerSocket.java:104)
	... 43 more
```

**Comment by Patras3** [comment]
Failed again on _5.3.z_ with corretto-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-CorrettoJDK17/68/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _master_ with oracle-11 (little-endian): https://jenkins.hazelcast.com/job/Hazelcast-master-little-endian/261/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _master_ with corretto-17: https://jenkins.hazelcast.com/job/Hazelcast-master-CorrettoJDK17/97/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by frant-hartm** [comment]
Failed on PR builder

https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/19416/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _5.3.z_ with microsoft-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-MicrosoftJDK17/42/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _master_ with OpenJDK11: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64/51/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _5.3.z_ with oracle-11: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-sonar/186/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _5.3.z_ with microsoft-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-MicrosoftJDK17/46/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with OpenJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK17-Esxi7/52/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _master_ with OpenJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK17-Esxi7/53/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _5.3.z_ with microsoft-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-MicrosoftJDK17/49/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with openjdk-ea (openjdk-21): https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK-ea/53/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by ihsandemir** [comment]
Another failure at PR build [here](https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/20743/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/).
```
java.io.UncheckedIOException: Failed to bind to /127.0.0.1:5001
	at com.hazelcast.internal.tpcengine.nio.NioAsyncServerSocket.bind(NioAsyncServerSocket.java:106)
	at com.hazelcast.internal.tpcengine.net.AsyncServerSocket.bind(AsyncServerSocket.java:102)
	at com.hazelcast.internal.tpcengine.net.AsyncServerSocketTest.test_createCloseLoop_withSameReactor(AsyncServerSocketTest.java:255)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:459)
	at java.base/sun.nio.ch.Net.bind(Net.java:448)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at com.hazelcast.internal.tpcengine.nio.NioAsyncServerSocket.bind(NioAsyncServerSocket.java:104)
	... 3 more
```

**Comment by Patras3** [comment]
Failed again on _master_ with OpenJDK11: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64/67/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with OpenJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK17-Esxi7/57/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _master_ with OpenJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK17-Esxi7/58/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with openjdk-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZuluJDK17/107/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _master_ with openjdk-17: https://jenkins.hazelcast.com/job/Hazelcast-master-ZuluJDK17/107/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with OpenJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK17-Esxi7/60/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _5.3.z_ with microsoft-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-MicrosoftJDK17/56/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _5.3.z_ with microsoft-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-MicrosoftJDK11/57/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by sumnerib** [comment]
Failed again on _master_ with OpenJDK11: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64/74/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _5.3.z_ with corretto-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-CorrettoJDK17/88/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by srknzl** [comment]
Failed on PR builder https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/21832/

**Comment by olukas** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/159/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _5.3.z_ with oracle-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-OracleJDK17/91/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _5.3.z_ with microsoft-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-MicrosoftJDK17/63/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _5.3.z_ with microsoft-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-MicrosoftJDK17/65/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by JackPGreen** [comment]
Failed in [PR builder](https://github.com/hazelcast/hazelcast-mono/pull/416#issuecomment-1891650433)

**Comment by Patras3** [comment]
Failed again on _master_ with microsoft-11: https://jenkins.hazelcast.com/job/Hazelcast-master-MicrosoftJDK11/99/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _master_ with oracle-11 (force-offload): https://jenkins.hazelcast.com/job/Hazelcast-master-force-offload/82/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by JackPGreen** [comment]
This [was not fixed](https://github.com/hazelcast/hazelcast-mono/pull/548#issuecomment-1914513879) by my changes.

**Comment by Patras3** [comment]
Failed again on _5.3.z_ with oracle-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-OracleJDK17/98/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by Patras3** [comment]
Failed again on _master_ with OpenJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK17-arm64/4/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with ZuluJDK21: https://jenkins.hazelcast.com/job/Hazelcast-master-ZuluJDK21/2/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by olukas** [comment]
Failed again on _master_ with CorrettoJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-CorrettoJDK17-arm64/3/testReport/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by JackPGreen** [comment]
If the problem is a ports already in use, and the port is hardcoded - wouldn't overlapping tests cause issues? Or if any _other_ test is also using `5000`?

**Comment by olukas** [comment]
@JackPGreen so the fix should be pick port number dynamically and not use hardcoded one, right?

**Comment by JackPGreen** [comment]
> @JackPGreen so the fix should be pick port number dynamically and not use hardcoded one, right?

Yes, that's my thinking. Just putting something together now to start the discussion.

**Comment by sumnerib** [comment]
Failed again on _master_ with oracle-17 (sonar build): https://jenkins.hazelcast.com/job/Hazelcast-master-sonar/407/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by AleksPeychev** [comment]
Failed again on _5.3.z_ with oracle-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-OracleJDK17/102/testReport/junit/com.hazelcast.internal.tpcengine.nio/NioAsyncServerSocketTest/test_createCloseLoop_withSameReactor/

**Comment by JackPGreen** [comment]
Fixed in `5.3.z` with [backport](https://github.com/hazelcast/hazelcast-mono/pull/916)
